py/src/braintrust/logger.py

```diff
@@ -2523,7 +2531,31 @@ def make_final_value_callback(v):
     return serializable_partial_record, lazy_partial_record
 
 
-class Dataset(ObjectFetcher):
+class Origin(TypedDict):
```

Comment:
Should we add comments linking these to the typespecs they are mirroring?

Reply:
I'm going to generate the typespecs automatically. Moved that to a separate PR. This PR will be focused on fixing the type annotation of Eval which will be easier to test.